mos_tests/neutron/python_tests/test_ovs_restart.py

```diff
@@ -480,3 +482,198 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
+    """Restart ovs-agents with iperf traffic background"""
+
+    def create_image(self, path, name):
+        """
+
+        :param path: path to folder where image_file located
+        :param name: name of file
+        :return: image object for Glance
+        """
+        full_path = os.path.join(path, name)
+        with open(full_path, 'rb') as image_file:
+            image = self.os_conn.glance.images \
+                .create(name="image_ubuntu",
+                        disk_format='qcow2',
+                        data=image_file,
+                        container_format='bare')
+        return image
+
+    def get_lost_percentage(self, output):
+        """
+
+        :param output: list of lines (output of iperf client)
+        :return: percentage of lost datagrams
+        """
+        lost_datagrams_rate_pattern = re.compile('\d+/\d+ \((\d+)%\)')
+        server_report_flag = False
+        for line in output:
+            if server_report_flag:
+                result = lost_datagrams_rate_pattern.search(line)
+                if result:
+                    return result.group(1)
+            elif line.endswith("Server Report:\n"):
+                server_report_flag = True
+        return False
+
+    def launch_iperf_server(self, vm, keypair, vm_login, vm_pwd):
+        """Launch iperf server"""
+        server_cmd = 'iperf -u -s -p 5002 </dev/null > ~/iperf.log 2>&1 &'
+        res_srv = self.run_on_vm(vm, keypair, server_cmd, vm_login=vm_login,
+                                 vm_password=vm_pwd)
+        return res_srv
+
+    def launch_iperf_client(self, client, server, keypair, vm_login,
+                            vm_pwd, background=False):
+        """Launch iperf client"""
+        client_cmd = 'iperf --port 5002 -u --client {0} --len 64' \
+                     ' --bandwidth 1M --time 60 -i 10' \
+            .format(self.os_conn.get_nova_instance_ips(server)['fixed'])
+        if background:
+            client_cmd = ' '.join(
+                (client_cmd, '< /dev/null > ~/iperf_client.log 2>&1 &'))
+        res = self.run_on_vm(client, keypair, client_cmd, vm_login=vm_login,
+                             vm_password=vm_pwd)
+        return res
+
+    @pytest.fixture(autouse=True)
+    def _prepare_openstack(self, init):
+        """Prepare OpenStack for scenarios run
+
+        Steps:
+            1. Update default security group
+            2. Upload the image with pre-installed iperf
+            3. Create router01, create networks net01: net01__subnet,
+            192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
+            attach them to router01.
+            4. Create keypair
+            5. Launch vm1 in net01 network and vm2 in net02 network
+            on different computes
+            6. Go to vm1 console and send pings to vm2
+        """
+
+        self.setup_rules_for_default_sec_group()
+        vm_image = self.create_image('/home/aallakhverdieva/images/',
```

Comment:
I see. In such case will be better to make it as a setting value (in mos-tests/settings.py) like other values. And add check, which skips this test, if image file not exists.
Also you may user `os.path.expanduser(~/images)` to get path relative to home folder.